flink-mesos/src/main/scala/org/apache/flink/mesos/scheduler/LaunchCoordinator.scala

```diff
@@ -70,7 +70,10 @@ class LaunchCoordinator(
             + s"of disk: ${lease.diskMB()} MB, network: ${lease.networkMbps()} Mbps")
           schedulerDriver.declineOffer(lease.getOffer.getId)
         }
-      }).build
+      })
+      // avoid situations where we have lots of expired offers and we only expire a couple at a time
+      .withRejectAllExpiredOffers()
```

Comment:
probably they will want to make this logic configurable, since caching could be a nice feature for some cases

Reply:
Yeah, we could do this via a config knob as well if needed.